File: packages/engine/src/cpus/6502/instructions/txs.js
import { CycleType } from '..'

/**
 * Transfer X to Stack pointer
 *
 * SP = X
 *
 * @example
 * MODE          SYNTAX        HEX LEN TIM
 * Implied       TXS           $9A  1   2
 */
export function* txs() {
  this.sp = this.x
  yield CycleType.GET
}

File: packages/engine/src/cpus/6502/instructions/txs.zig
const std = @import("std");
const CPU = @import("../cpu.zig").CPU;
const Addressing = @import("../opcode.zig").Addressing;

// Transfer X to Stack pointer
//
// SP = X
//
// MODE          SYNTAX        HEX LEN TIM
// Implied       TXS           $9A  1   2

pub fn txs(self: *CPU, addressing: Addressing) void {
    _ = addressing;
    self.sp = self.x;
}
